### PR TITLE
[ODS-5888] External cache with encryption

### PR DESCRIPTION
**Pull request by simpat-jesus**

### CODE REVIEW

File: Application/EdFi.Ods.Features/ExternalCache/ExternalCacheProvider.cs [hunk 1]
```diff
@@ -169,7 +169,8 @@ private object Deserialize(string @string)
 
             try
             {
-                return JsonConvert.DeserializeObject(@string, _defaultSerializerSettings);
+                // JsonConvert.DeserializeObject without a Type, returns a JObject, that will fail to be casted to a DescriptorMaps object
```
Comment (simpat-jesus) [inline Application/EdFi.Ods.Features/ExternalCache/ExternalCacheProvider.cs:172]: JsonConvert.DeserializeObject returns a JObject that will throw an exception when trying to cast it to DescriptorMaps.

**Review by simpat-adam — APPROVED** [review]

File: Application/EdFi.Ods.Features/ExternalCache/ExternalCacheProvider.cs (hunk 1 above)
Comment (gmcelhanon) [reply Application/EdFi.Ods.Features/ExternalCache/ExternalCacheProvider.cs:172]: How is this change rationalized against the comment on [Line 90](https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS/pull/765/files#diff-f532e1bea12f11735b35bffb23c070c38888d4823ed590b96deebc861bb7a70bR90) above? What has this changed?

File: Application/EdFi.Ods.Api/Container/Modules/ApplicationModule.cs [hunk 2]
```diff
@@ -195,6 +196,10 @@ protected override void Load(ContainerBuilder builder)
                 .EnableInterfaceInterceptors()
                 .SingleInstance();
 
+            builder.RegisterType<ApiClientDetailsCacheKeyProvider>()
```
Comment (gmcelhanon) [inline Application/EdFi.Ods.Api/Container/Modules/ApplicationModule.cs:199]: This class is now only used by the ExternalCache feature:
image

So let's move the interface and implementation classes (`IApiClientDetailsCacheKeyProvider` and `ApiClientDetailsCacheKeyProvider`) to the ExternalCache folder in the Features project, and also move this registration should move to the `ExternalCacheModule` class.

**Review by gmcelhanon — CHANGES_REQUESTED** [review]

File: Application/EdFi.Ods.Features/ExternalCache/ExternalCacheProvider.cs (hunk 1 above)
Comment (simpat-jesus) [reply Application/EdFi.Ods.Features/ExternalCache/ExternalCacheProvider.cs:173]: ExternalCacheProvider is used in 3 overrides

PersonUniqueIdToUsiCache -> deserializes to IdentityValueMaps
ContextualCachingInterceptor -> registered to intercept IDescriptorMapsProvider, deserializes to DescriptorMaps
CachingApiClientDetailsProviderDecorator -> deserializes to ApiClientDetails

**Review by gmcelhanon — APPROVED** [review]

**Pull request merged**